src/transformers/pipelines/mask_generation.py

```diff
@@ -0,0 +1,288 @@
+                inference_context = self.get_inference_context()
+                with inference_context():
+                    model_inputs = self._ensure_tensor_on_device(model_inputs, device=self.device)
+                    image_embeddings = self.model.get_image_embeddings(model_inputs.pop("pixel_values"))
+                    model_inputs["image_embeddings"] = image_embeddings
+
+        if points_per_batch > 1:
+            for i in range(0, grid_points.shape[1], points_per_batch):
+                batched_points = grid_points[:, i : i + points_per_batch, :, :]
+                labels = input_labels[:, i : i + points_per_batch]
+                is_last = i == grid_points.shape[1] - points_per_batch
+                yield {
+                    "input_points": batched_points,
+                    "input_labels": labels,
+                    "input_boxes": crop_boxes,
+                    "is_last": is_last,
+                    **model_inputs,
+                }
+        else:
+            yield {
+                "input_points": grid_points,
+                "input_labels": input_labels,
+                "input_boxes": crop_boxes,
+                "is_last": True,
+                **model_inputs,
+            }
+
+    def _forward(
+        self,
+        model_inputs,
+        pred_iou_thresh=0.88,
+        stability_score_thresh=0.95,
+        mask_threshold=0,
+        stability_score_offset=1,
+    ):
+        input_boxes = model_inputs.pop("input_boxes")
+        is_last = model_inputs.pop("is_last")
+        original_sizes = model_inputs.pop("original_sizes")
+        reshaped_input_sizes = model_inputs.pop("reshaped_input_sizes")
+
+        model_outputs = self.model(**model_inputs)
+
+        # post processing happens here in order to avoid CPU GPU copies of ALL the masks
+        low_resolution_masks = model_outputs["pred_masks"]
+        masks = self.image_processor.post_process_masks(
+            low_resolution_masks, original_sizes, reshaped_input_sizes, mask_threshold, binarize=False
+        )
+        iou_scores = model_outputs["iou_scores"]
+        masks, iou_scores, boxes = self.image_processor.filter_masks(
+            masks[0],
+            iou_scores[0],
+            original_sizes[0],
+            input_boxes[0],
+            pred_iou_thresh,
+            stability_score_thresh,
+            mask_threshold,
+            stability_score_offset,
+        )
+        return {
+            "masks": masks,
+            "is_last": is_last,
+            "boxes": boxes,
+            "iou_scores": iou_scores,
+        }
+
+    def postprocess(
+        self,
+        model_outputs,
+        output_rle_mask=False,
+        output_bboxes_mask=False,
+        crops_nms_thresh=0.7,
+    ):
+        all_scores = []
+        all_masks = []
+        all_boxes = []
+        for model_output in model_outputs:
+            all_scores.append(model_output.pop("iou_scores"))
+            all_masks.extend(model_output.pop("masks"))
```

Comment:
The reason behind that is that `model_output.pop("masks")` returns a list of masks, and `post_process_for_mask_generation` expects a single list of masks instead of nested lists. Therefore you need to call `extend`